numpy/core/tests/test_umath.py

```diff
@@ -526,6 +527,21 @@ def test_integer_to_negative_power(self):
             assert_raises(ValueError, np.power, one, b)
             assert_raises(ValueError, np.power, one, minusone)
 
+    @pytest.mark.skipif(not hasattr(np, 'float128'),
+                        reason='test requires float128')
+    def test_float128(self):
+        x = np.array([0, 1, 2], dtype=np.float128)
+        # On Mac OSX, this would trigger a divide-by-zero warning.
+        y = np.power(x, 4)
+        assert_array_equal(y, [0.0, 1.0, 16.0])
+
+    @pytest.mark.skipif(not sys.platform == 'darwin',
+                        reason='only test on Mac OSX (i.e. darwin)')
+    def test_float128_precision(self):
+        # Check that we are using more than double precision.
+        v = np.float128(2) ** 16383
+        assert_allclose(v, np.float128('5.9486574767861588254e+4931'))
+
```

Comment:
Do we expect the second test to fail on any platform with 'float128'? I.e., why not `skipif(not hasattr(np, 'float128')`,

Reply:
It is possible that there are (or will be) other platforms that have also implemented `power` for `float128` by falling back to the double precision version.  Currently, this test will fail if it is run on the `ppc64le` platform on Travis-CI.  So for now, the test is restricted to Mac OSX/Darwin.

Reply:
As long as we are fixing, could we fix it for ppy64le as well?

Reply:
I started working on this because I have a Mac, so it seemed like it would be straightforward to fix.  The `feclearexcept(FE_ALL_EXCEPT)` hack shows that, even on the platform for which the problem was originally reported, it is not as easy as it seemed it should be.  Without having a `ppc64le` computer to work on directly, testing and debugging means lots of guess work, and repeatedly running tests via our CI suite.  That's pretty inefficient, and probably not a good use of time and computer resources.

Reply:
> Without having a ppc64le computer to work on directly,

You should be eligible for the gcc compile farm machines, which do include `ppc64le` machines that are suitable for iterative work, see here: https://cfarm.tetaneutral.net/users/new/

I've been using them & they are pretty great machines.